tests/util/types.js

```diff
@@ -0,0 +1,135 @@
+const types = {
+	any: {
+		syntax: "*",
+		initialValue: "0",
+		values: ["foo", "42"],
+	},
+
+	integer: {
+		initialValue: "0",
+		values: ["42", "-42"],
+		property: "z-index",
+	},
+
+	number: {
+		initialValue: "0",
+		values: ["42", "42.42", "-0.1", "calc(infinity)"],
+		property: "opacity",
+	},
+
+	length: {
+		initialValue: "0px",
+		values: ["42px", "-42px"],
+		property: "width",
+	},
+
+	percentage: {
+		initialValue: "0%",
+		values: ["42%", "-42%"],
+		property: "opacity", // actually <number> | <percentage> but there is no property that only does <percentage>
+	},
+
+	"length-percentage": {
+		initialValue: "0%",
+		values: ["42px", "-42px", "42%", "-42%"],
+		property: "width",
+	},
+
+	color: {
+		initialValue: "rgba(0, 0, 0, 0)",
+		values: ["rgb(0, 0, 0)", "hsl(0, 100%, 50%)"],
+		property: "color",
+	},
+
+	angle: {
+		initialValue: "0deg",
+		values: ["0.5turn", "360deg"],
+		property: "rotate",
+	},
+
+	resolution: {
+		initialValue: "96dpi",
+		values: ["42dppx", "42dpi"],
+		property: "resolution",
+	},
+
+	time: {
+		initialValue: "0s",
+		values: ["42s", "42ms"],
+		property: "transition-duration",
```

Comment:
I thought we couldn't observe the `transition` property. This test will always time out.

Reply:
OMG yes, I'm an idiot lol. I guess we'll need to remove that line then.